File: app/src/main/java/i_Java_Interop/KotlinFromJava.java
package i_Java_Interop;

public class KotlinFromJava {

    void demoKotlinFromJava() {

        new Foo(1);
        new Foo(1, 2);

        new Foo(1).f("a");
        new Foo(1).f("a", 1);
        new Foo(1).f("a", 1, "c");
        new Foo(1).f("a", 1, "c");


    }
}

File: app/src/main/java/i_Java_Interop/KotlinFromJava.kt
package i_Java_Interop

class Foo @JvmOverloads constructor(x: Int, y: Double = 0.0) {

    @JvmOverloads
    fun f(a: String, b: Int = 0, c: String = "abc", d : String = "d") { /**/

    }
}
